### DISCUSSION TITLE
📢 Who uses Classic.Avalonia

### DISCUSSION THREAD

**Discussion by BAndysc** [BAndysc/Classic.Avalonia#2] [Show and tell]
If you've used `Classic.Avalonia` in your project, feel free to share screenshots and links to your app here. Let's embrace the classic look and feel together! Even if this is just a sample demo app, share it here :)

**Comment by rabbitism** [comment]
FYI Ursa.Avalonia is a look-less UI library, we would like to adopt more themes. Classic might be the next one (not open source thou). 
![image](https://github.com/user-attachments/assets/eddc2499-9494-46c8-bc79-8b40adeb9f3b)

**Comment by ykarpeev** [comment]
I used it for a small pdf converter GUI I made.

![image](https://github.com/user-attachments/assets/27c2484c-7795-45aa-bc00-ffb859636eb1)

**Reply by BAndysc** [reply]
I love it, looks like it was taken straight from 1999!

**Comment by djeikyb** [comment]
This theme is gorgeous. It's remarkable how helpful it is prototyping. With the default fluent theme, nothing looks quite right without A LOT of fiddling. It's a bit too blank canvas. Even though I'm on a mac, and the win9x theme is very out of place, there's a clear design language to follow and it's easier to get on with trying out application design.

**Reply by BAndysc** [reply]
Absolutely agree! Classic look and feel is all about productivity and clear separation of what is interactive and what is not! Thanks for your words.

**Comment by djeikyb** [comment]
Here's a tiny little app I made for me and one other person. Tryna make it easy for them to fetch commits and build and run a project even though they aren't a dev and aren't savvy with dev tooling.

![win9x-trunkflight](https://github.com/user-attachments/assets/8bcc0e38-6f13-46cf-bfae-13f26456b04b)

**Comment by BAndysc** [comment]
I will use my thread to showcase my other app with Classic Theme :)
![image](https://github.com/user-attachments/assets/caa66949-6885-4ea6-864a-e95cc0cd65cb)

[WoW Database Editor](https://github.com/BAndysc/WoWDatabaseEditor) is an IDE for the TrinityCore MMORPG server. The default theme is a modern, Windows 11-esque style, but a classic Windows 9x theme is also available..

**Comment by hacklex** [comment]
I realized there's no free, ad-free, fully offline stand-alone simple image editor that could be used to make pixel art-style icons, on Mac. So I asked JetBrains AI Assistant to write one for me. I only made pretty GUI and fixed several logic flaws -- most of the code was already fine.

I went to great lengths though to make `Padding`s and `Margin`s look well enough to almost obscure the fact that the whole thing is built on Avalonia, which traditionally had problems with text baseline alignment -- from the days when it was called perspex, till 2025 -- and changing anything at this point would mean breaking everything for everyone who carefully prepared their styles to account for every piece of text drawn a pixel or two above its natural position.

So, a [simple retro image editor](https://github.com/hacklex/ClassicPixel) (so far it only has PNG support) was born.
![image](https://github.com/hacklex/ClassicPixel/raw/master/Assets/Screenshot.png)


...While building it, I realized that many peculiar issues with button contents being offset inconsistently on MouseDown for different controls (and sometimes even for exactly identical buttons -- only because one contained an Image, and the other didn't) -- are caused by the fact that ClassicBorderDecorator tries to automatically offset its content according to its BorderStyle value. This results in various buttons offsetting their content by different values (some move the content straight down, some straight to the right, and some diagonally -- and some probably fail to move it at all). I think this begs for some refactoring on the library level, not in the user code. Also, as the decorator is heavily used outside buttons and similar active elements, the whole idea does not look very good to me. Rather, I would separate the offsetting logic into a special control that has it as its sole purpose, and does nothing else. Single responsibility principle, right? The BorderDecorator decorates, the ContentOffsetter offsets its content. Way earlier, I already created [an issue on the matter](https://github.com/BAndysc/Classic.Avalonia/issues/10), but back then we failed to pinpoint its root I think, and it got closed. Would be good if we addressed it after all.

Next, I couldn't manage to select the small icon from the ICO file for the `TitleBar` while keeping the large icon for the system taskbar, using the standard ClassicWindow class. Had to draw the entire window on my own. An issue [is of course already open](https://github.com/BAndysc/Classic.Avalonia/issues/17).

Next, perhaps I just don't know it but is there a button-like style for radio buttons?

**Comment by teobugslayer** [comment]
I am maintaining a small Avalonia / NativeAOT demo, which other people can use as a starting point for their own Avalonia-based projects.

I've finally got the time to integrate the Classic theme. The project is [here](https://github.com/teobugslayer/AvaloniaCoreRTDemo).

The glory of the classic UI:

<img width="1125" height="893" alt="{7C738FCB-FCBB-40D6-A9D8-E97E5E86B72C}" src="https://github.com/user-attachments/assets/8f0f8840-b352-4e4b-9c4d-ecf65ca52dba" />

**Comment by nilaoda** [comment]
https://github.com/nilaoda/TSCutter.GUI

[TSCutter.GUI](https://github.com/nilaoda/TSCutter.GUI) is a lightweight tool designed to efficiently cut TS (Transport Stream) video files — with keyframe-accurate trimming, zero re-encoding, and cross-platform support (Windows, Linux, macOS).  

I’ve also enhanced the UI with a **native dark mode** that automatically adapts to system preferences, preserving the classic desktop aesthetic while reducing eye strain during long editing sessions.

<img width="1760" height="1340" alt="light" src="https://github.com/user-attachments/assets/46461cbc-734a-4da9-9941-7d5b89b3460c" />

**Reply by BAndysc** [reply]
It looks phenomenal! I love the use of the toolbar and the icons, and I really like the dark theme! Would you like to create a PR with this dark style? I think it's worth merging!

**Reply by nilaoda** [reply]
> It looks phenomenal! I love the use of the toolbar and the icons, and I really like the dark theme! Would you like to create a PR with this dark style? I think it's worth merging!

I’d love to contribute this to your repository so others can benefit from it — your work is truly excellent, and I’m grateful for the foundation you’ve built. Right now, my implementation is still a bit rough around the edges, but I’m actively working to improve it, and I’ll open a PR once it’s ready.

**Comment by Neonalig** [comment]
Was tired of having random spouts of dead air when using the computer and figured it would be nice to have a little Internet Radio utility that lives in the system tray... a [Traydio](https://github.com/Neonalig/Traydio). Eventually I want to hook it up to an Elgato Stream Deck custom extension too, but that's future me's problem.

<table>
  <tr>
    <td align="center">
      <img src="https://github.com/user-attachments/assets/2f79a2f5-21cd-4d41-8f15-8ce79e21a174" alt="Traydio stations window" width="420" />
      <br />
      <sub>Station Manager</sub>
    </td>
    <td align="center">
      <img src="https://github.com/user-attachments/assets/5dbaddcf-0d15-4266-8e64-ff684a25ec60" alt="Traydio station search" width="420" />
      <br />
      <sub>Search</sub>
    </td>
  </tr>
  <tr>
    <td align="center">
      <img src="https://github.com/user-attachments/assets/bef979b3-e7fa-447f-aa56-b1251ea4f336" alt="Traydio about window" width="420" />
      <br />
      <sub>About</sub>
      <br />
      <sub>Had to add a settings button to let users download WMIC as its no longer included in Windows 11 :)</sub>
    </td>
    <td align="center">
      <img src="https://github.com/user-attachments/assets/cba7c5f9-8996-4006-a385-3d280eedb443" alt="Traydio system tray" width="420" />
      <br />
      <sub>System Tray</sub>
    </td>
  </tr>
</table>

**Reply by BAndysc** [reply]
Nice! Love it!
